openpype/pipeline/load/plugins.py

```diff
@@ -79,6 +82,36 @@ def apply_settings(cls, project_settings, system_settings):
                 print("  - setting `{}`: `{}`".format(option, value))
             setattr(cls, option, value)
 
+    @classmethod
+    def is_compatible_loader(cls, context):
+        """Return whether a loader is compatible with a context.
+
+        This checks the version's families and the representation for the given
+        Loader.
+
+        Returns:
+            bool
+        """
+
+        maj_version, _ = schema.get_schema_version(context["subset"]["schema"])
+        if maj_version < 3:
+            families = context["version"]["data"].get("families", [])
+        else:
+            families = context["subset"]["data"]["families"]
+
+        representation = context["representation"]
+        has_family = (
+            "*" in cls.families or any(
+                family in cls.families for family in families
+            )
+        )
+        representations = cls.get_representations()
+        has_representation = (
+            "*" in representations
+            or representation["name"] in representations
+        )
```

Comment:
Since we're tweaking anyway - I think we could optimize this by doing the lookup against a set.
This should make quite a difference for plugins with many valid representations but the context not being in it or for an instance with many families against a plugin with many valid families this should also speed things up.

```suggestion
        plugin_families = frozenset(cls.families)
        has_family = (
            "*" in plugin_families or any(
                family in plugin_families for family in families
            )
        )
        if not has_family:
            return False
        
        representation = context["representation"]
        representations = frozenset(cls.get_representations())
        has_representation = (
            "*" in representations
            or representation["name"] in representations
        )
        return has_representation
```
This also opts out from the representation check if the family already didn't pass.